### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by mikejsavage** [michaelforney/samurai#12]
Cool project, the problem I have with ninja is you can't static link with musl, and samurai solves that! I'm hoping to have a build system I can add to the repo and forget about so people can just clone + build and it will always work.

I need Windows support, so if I work on that would you be interested in merging it to the main repo?

I took a quick look and it seems like there's not much that needs changing. File I/O stuff and the job spawning/monitoring.

**Comment by michaelforney** [comment]
Yes, I would be interested in merging Windows support. I already have some WIP changes to the lexer and deps/build log to use only C99 library functions.

As you noted, job spawning/monitoring is really the only thing that needs special OS-specific code. I think the best way to do this is to define a generic interface (maybe in an `os.h` header), and then provide implementations for each OS (e.g. `posix.c` and `windows.c`). I definitely do *not* want Windows support added through ifdefs in `build.c`.

> Cool project, the problem I have with ninja is you can't static link with musl, and samurai solves that!

BTW, what issue do you have statically linking ninja with musl? I've done it in the past without issues (before I wrote samurai).

**Comment by mikejsavage** [comment]
Great, would it better if I wait for those changes to go in then? (or post the diff atm and I can take over)

> BTW, what issue do you have statically linking ninja with musl? I've done it in the past without issues (before I wrote samurai).

When I tried I ran into problems with my C++ headers relying on glibc things

**Comment by michaelforney** [comment]
FYI, Zach van Rijn is providing static ninja binaries for Linux and Windows:
https://groups.google.com/d/msg/ninja-build/tPFgdGOAPo8/syQbQILxEQAJ

**Comment by RoyiAvital** [comment]
Is there any progress on that?
Namely having a Samurai binary for Windows?

**Comment by michaelforney** [comment]
The most recent effort was in https://github.com/michaelforney/samurai/pull/37.

I don't have any machines that run Windows, so it's up to whoever wants to do the work.
